enterprise_gateway/services/processproxies/processproxy.py

```diff
@@ -53,7 +53,24 @@
 remote_user = os.getenv('EG_REMOTE_USER', getpass.getuser())
 remote_pwd = os.getenv('EG_REMOTE_PWD')  # this should use password-less ssh
 
-local_ip = localinterfaces.public_ips()[0]
+
+# Allow users to specify local ips that should not be included when determining the response
+# address.  For example, on systems with many network interfaces, some may have their IPs appear
+# the local interfaces list (e.g., docker0 (172.17.0.1) is an example) that should not be used.
+# This env can be used to indicate such IPs.
+local_ip_blacklist = os.getenv('EG_LOCAL_IP_BLACKLIST','').split(',')
+
+
+def get_local_ip():
+    """
+    Honor the blacklist-ed IPs, locating the first not in the list.
+    """
+    for ip in localinterfaces.public_ips():
+        if ip not in local_ip_blacklist:
```

Comment:
I am not sure this would be flexible enough for virtual/containerized environments. If we could use a IP class, or something like a regEx, it would give the flexibility we need (e.g. 172.* instead of a specific ip as it might change from one node versus another particularly when we start launching EG on K8s, Docker, or in HA). Thoughts ?

Reply:
Having each item of the comma separated list be a regex is a good idea. However, even in the case being handled 172.* wasn't sufficient, but 172.17.* would be.
I think we should be careful not to over engineer this since use of this env should be rare. Spectrum happens to have these hybrid-visible networks.
I can add regex support tomorrow. Setting an env with regex characters may be tricky, but I'm sure it can be made to work.

Reply:
@lresende - Just pushed a commit (amending the original) with support for regular expressions.  Thanks for pointing that out - that's a great change.